File: src/CSharp/GeometricObjects/PlanesIntersection.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Windows.Forms;
using System.Diagnostics;

using Kitware.VTK;

namespace ActiViz.Examples {
   public partial class Form1 : Form {
      public Form1() {
         InitializeComponent();
      }


      private void renderWindowControl1_Load(object sender, EventArgs e) {
         try {
            PlanesIntersection();
         }
         catch(Exception ex) {
            MessageBox.Show(ex.Message, "Exception", MessageBoxButtons.OK);
         }
      }


      private void PlanesIntersection() {
         vtkSphereSource sphereSource = vtkSphereSource.New();
         sphereSource.Update();

         double[] bounds = new double[6];
         bounds = sphereSource.GetOutput().GetBounds();

         vtkPoints box = vtkPoints.New();

         box.SetNumberOfPoints(8);

         double xMin, xMax, yMin, yMax, zMin, zMax;
         xMin = bounds[0]; xMax = bounds[1];
         yMin = bounds[2]; yMax = bounds[3];
         zMin = bounds[4]; zMax = bounds[5];

         box.SetPoint(0, xMax, yMin, zMax);
         box.SetPoint(1, xMax, yMin, zMin);
         box.SetPoint(2, xMax, yMax, zMin);
         box.SetPoint(3, xMax, yMax, zMax);
         box.SetPoint(4, xMin, yMin, zMax);
         box.SetPoint(5, xMin, yMin, zMin);
         box.SetPoint(6, xMin, yMax, zMin);
         box.SetPoint(7, xMin, yMax, zMax);

         vtkPlanesIntersection planesIntersection = vtkPlanesIntersection.New();
         planesIntersection.SetBounds(bounds[0], bounds[1], bounds[2], bounds[3], bounds[4], bounds[5]);
         int intersects = planesIntersection.IntersectsRegion(box);
         Debug.WriteLine("Intersects? " + ( ( intersects == 1 ) ? true : false ).ToString());
         // nothing to visualize
      }
   }
}

File: src/Java/GeometricObjects/PlanesIntersection.java
import vtk.vtkSphereSource;
import vtk.vtkPoints;
import vtk.vtkPlanesIntersection;
import vtk.vtkNativeLibrary;

class PlanesIntersection{  
	
	static {
	    if (!vtkNativeLibrary.LoadAllNativeLibraries()) {
	      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) {
	        if (!lib.IsLoaded()) {
	          System.out.println(lib.GetLibraryName() + " not loaded");
	        }
	      }
	    }
	    vtkNativeLibrary.DisableOutputWindow(null);
	  }
	
	
    public static void main(String args[]){  
    	
    	vtkSphereSource SphereSource = new vtkSphereSource();
		  SphereSource.Update();
		  
		  double bounds[] = new double[6];
		  		  
		  SphereSource.GetOutput().GetBounds(bounds);
		  vtkPoints Box = new vtkPoints();
		  Box.SetNumberOfPoints(8);
		  
		  double xMin, xMax, yMin, yMax, zMin, zMax;
		  xMin = bounds[0]; xMax = bounds[1];
		  yMin = bounds[2]; yMax = bounds[3];
		  zMin = bounds[4]; zMax = bounds[5];

		  
		  Box.SetPoint(0, xMax, yMin, zMax);
		  Box.SetPoint(1, xMax, yMin, zMin);
		  Box.SetPoint(2, xMax, yMax, zMin);
		  Box.SetPoint(3, xMax, yMax, zMax);
		  Box.SetPoint(4, xMin, yMin, zMax);
		  Box.SetPoint(5, xMin, yMin, zMin);
		  Box.SetPoint(6, xMin, yMax, zMin);
		  Box.SetPoint(7, xMin, yMax, zMax);
		  
		  
		  vtkPlanesIntersection PlanesIntersection = new vtkPlanesIntersection();
		  PlanesIntersection.SetBounds(bounds);
		  
		  int intersects = PlanesIntersection.IntersectsRegion(Box);
		  System.out.print("Intersects:" + intersects );
		  
    }  
}   
